airflow/providers/common/sql/hooks/sql.py

```diff
@@ -594,7 +594,7 @@ def insert_rows(
                             conn.commit()
                             self.log.info("Loaded %s rows into %s so far", i, table)
                     conn.commit()
-        self.log.info("Done loading. Loaded a total of %s rows into %s", len(rows), table)
+        self.log.info("Done loading. Loaded a total of %s rows into %s", len(list(rows)), table)
```

Comment:
This will always be 0 because the rows would have been consumed when they are inserted into the database.

Reply:
Should I create an old school increment in the two for loop or is there any better/pythonic way?

Reply:
yes. Best is to return count from both variants - seems to be best way.

Reply:
Sorry my bad indeed I didn’t know why they used an old school counter. Maybe add a unit test then to make sure that case is also covered?

Reply:
That's a good point. I'll do that on Monday

Reply:
If you want I can also add it on monday as I caused this mess anyway.

Reply:
> If you want I can also add it on monday as I caused this mess anyway.

Don't feel bad about it :) You could not know that `rows` could also be a generator. But I take the offer with pleasure :) I'll merge this one once I fix the increment, then I'll leave you the unit test to add :)

Reply:
I fixed the increment. Feel free to re-review. Thanks!